source/features/restore-file.tsx

```diff
@@ -79,32 +87,19 @@ async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButt
 
 	filesRestored.add(menuItem);
 
-	menuItem.textContent = 'Restoring…';
-	event.preventDefault();
-	event.stopPropagation();
-
 	try {
 		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
-		const file = await getFile(filePath);
-
-		if (!file) {
-			// The file was created by this PR.
-			// This code won’t be reached if `highlight-deleted-and-added-files-in-diffs` works.
-			showError(menuItem, 'Nothing to restore. Delete file instead');
-			return;
-		}
-
-		if (file.isTruncated) {
-			showError(menuItem, 'Restore failed: File too big');
-			return;
-		}
-
-		await commitFileContent(menuItem, file.text, filePath);
+		// Show toast while restoring
+		await showToast(async progress => restoreFile(progress!, menuItem, filePath), {
+			message: 'Restoring…',
+			doneMessage: 'Restored!',
+		});
 
 		// Hide file from view
 		menuItem.closest('.file')!.remove();
 	} catch (error: unknown) {
-		showError(menuItem, 'Restore failed. See console for details');
+		menuItem.disabled = true;
+		menuItem.style.background = 'none'; // Disables hover background color
```

Comment:
I think we can leave them alone now
```suggestion
```

For example it would allow the user to update the token in the options and retry without refreshing the page

Reply:
Alright!